presto-hive-metastore/src/main/java/com/facebook/presto/hive/HdfsEnvironment.java

```diff
@@ -69,6 +72,9 @@ public ExtendedFileSystem getFileSystem(String user, Path path, Configuration co
     {
         return hdfsAuthentication.doAs(user, () -> {
             FileSystem fileSystem = path.getFileSystem(configuration);
+            if (isFileSystemDisabledCache(path, configuration)) {
```

Comment:
AFAIK, whether the filesystem is to be cached is checked in the method [FileSystem.get](https://github.com/apache/hadoop/blob/d4c8d4d4d203c934e8074b31289a28724c0842cf/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L367) method (which is called `path.getFilesystem(configuration`). Do we actually need this check?

Reply:
If the filesystem is disabled cache, then the filesystem is not wrapper by HadoopExtendedFileSystem, then checkState of filesystem is instance of  ExtendedFileSystem will aways failed.